tlsutil/config.go

```diff
@@ -455,12 +456,23 @@ func newX509CertPool(groups ...[]string) (*x509.CertPool, error) {
 			}
 		}
 	}
-	if len(pool.Subjects()) == 0 {
+	if !haveSubjects(pool) {
 		return nil, nil
 	}
 	return pool, nil
 }
 
+// getSubjects uses the reflect package to test for rawSubject fields.
+func haveSubjects(cp *x509.CertPool) bool {
+	subjectsIter := reflect.ValueOf(cp).Elem().FieldByName("byName").MapRange()
```

Comment:
this feels like it could panic if there is no field `byName` right?

Reply:
Yeah, that's a good call. I had a previous comment about how I needed to check something but wasn't familiar enough with `reflect`. I think I know how to handle it now.

Reply:
Actually this is the same code block RB pointed out above with my proposed soln. Will mark this resolved and we can move the discussion there.